proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,358 @@
+
+This feature is for dotnet.exe users.
+
+## Goals
+Design and implement `dotnet nuget config` command.
+
+## Non-Goals
+Design and implement in the same way for NuGet.exe command.
+
+## Solution
+The following command will be implemented in the `dotnet.exe` CLI.
+
+### `dotnet nuget config`
+
+#### Commands
+
+- Path
+
+List all the paths of NuGet configuration files that will be applied, when invoking NuGet command from the current working directory path.
+
+The listed NuGet configuration files are in priority order. So the order of loading those configurations is reversed, that is, loading order is from the bottom to the top. So the configuration on the top will apply.
+You may refer to [How settings are applied](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#how-settings-are-applied) for more details. 
+
+
+#### Options
+
+- -w|--working-dir <WORKING_DIRECTORY>
+
+Run this command as if working directory is set to the specified directory.
+
+If the specified `--working-dir` doesn't exist, an error is displayed indicating the `--working-dir` doesn't exist.
+
+> [!Note]
+> If `--working-dir` (or its parent directories) is not accessible, the command will ignore any NuGet configuration files under those directories without any warning/error. This is aligned with other NuGet commands.
+
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+#### Examples
+
+- List all the NuGet configuration file that will be applied, when invoking NuGet command in the current directory.
+
+```
+dotnet nuget config path 
+
+C:\Test\Repos\Solution\NuGet.Config
+C:\Test\Repos\NuGet.Config
+C:\Test\NuGet.Config
+C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config
+```
+
+- List all the paths of NuGet configuration files that will be applied, when invoking NuGet command in the specific directory.
+
+```
+dotnet nuget config path  --working-dir C:\Test\Repos
+
+C:\Test\Repos\NuGet.Config
+C:\Test\NuGet.Config
+C:\Users\username\AppData\Roaming\NuGet\NuGet.Config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
+C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config
+```
+
+- List all the NuGet configuration file that will be applied, but passing a non-exsiting `--working-dir`.
+
+```
+dotnet nuget config path  --working-dir C:\Test\NonExistingRepos
+
+Error: The path "C:\Test\NonExistingRepos" doesn't exist.
+```
+
+- List all the NuGet configuration file that will be applied, but passing an inaccessible `--working-dir`: C:\Test\AccessibleRepos\NotAccessibleSolution. 
+
+The configuration file under C:\Test\AccessibleRepos\NotAccessibleSolution\NuGet.Config will be ignored without any warning or error.
+
+```
+dotnet nuget config list  --working-dir C:\Test\AccessibleRepos\NotAccessibleSolution
```

Comment:
You probably just missed changing the list argument here.

Reply:
Good catch! Thanks! Fixed.